test/Microsoft.AspNetCore.Mvc.Core.Test/AcceptedAtActionResultTests.cs

```diff
@@ -0,0 +1,183 @@
+                actionName: expectedUrl,
+                controllerName: null,
+                routeValues: null,
+                value: null);
+
+            result.UrlHelper = urlHelper;
+            await result.ExecuteResultAsync(actionContext);
+
+            // Assert
+            Assert.Equal(StatusCodes.Status202Accepted, httpContext.Response.StatusCode);
+            Assert.Equal(expectedUrl, httpContext.Response.Headers["Location"]);
+        }
+
+        [Fact]
+        public async Task AcceptedAtActionResult_ThrowsOnNullUrl()
+        {
+            // Arrange
+            var httpContext = GetHttpContext();
+            var actionContext = GetActionContext(httpContext);
+            var urlHelper = GetMockUrlHelper(returnValue: null);
+
+            var result = new AcceptedAtActionResult(
+                actionName: null,
+                controllerName: null,
+                routeValues: null,
+                value: null);
+
+            result.UrlHelper = urlHelper;
+
+            // Act & Assert
+            await ExceptionAssert.ThrowsAsync<InvalidOperationException>(
+                async () => await result.ExecuteResultAsync(actionContext),
+            "No route matches the supplied values.");
+        }
+
+        private static ActionContext GetActionContext(HttpContext httpContext)
+        {
+            var routeData = new RouteData();
+            routeData.Routers.Add(Mock.Of<IRouter>());
+
+            return new ActionContext(httpContext,
+                                    routeData,
+                                    new ActionDescriptor());
+        }
+        private static HttpContext GetHttpContext()
+        {
+            var httpContext = new DefaultHttpContext();
+            httpContext.Request.PathBase = new PathString("");
+            httpContext.Response.Body = new MemoryStream();
+            httpContext.RequestServices = CreateServices();
+            return httpContext;
+        }
+
+        private static IServiceProvider CreateServices()
+        {
+            var options = new TestOptionsManager<MvcOptions>();
+            options.Value.OutputFormatters.Add(new StringOutputFormatter());
+            options.Value.OutputFormatters.Add(new JsonOutputFormatter(
+                new JsonSerializerSettings(),
+                ArrayPool<char>.Shared));
+
+            var services = new ServiceCollection();
+            services.AddSingleton(new ObjectResultExecutor(
+                options,
+                new TestHttpResponseStreamWriterFactory(),
+                NullLoggerFactory.Instance));
+
+            return services.BuildServiceProvider();
+        }
+
+        private static IUrlHelper GetMockUrlHelper(string returnValue)
+        {
+            var urlHelper = new Mock<IUrlHelper>();
+            urlHelper.Setup(o => o.Action(It.IsAny<UrlActionContext>())).Returns(returnValue);
+
+            return urlHelper.Object;
+        }
+
+        private class ResourceStatusBody
```

Comment:
Do you need a custom type? Can you just have it return a `new object()` in `TestData`